hydra-node/exe/hydra-node/Main.hs

```diff
@@ -31,23 +33,48 @@ import Hydra.Node (
   initEnvironment,
   runHydraNode,
  )
-import Hydra.Options (ChainConfig (..), LedgerConfig (..), Options (..), parseHydraOptions)
+import Hydra.Options (
+  ChainConfig (..),
+  Command (Publish, Run),
+  LedgerConfig (..),
+  Options (..),
+  PublishOptions (..),
+  parseHydraCommand,
+ )
 import Hydra.Party (Party)
 
 main :: IO ()
 main = do
-  o@Options{verbosity, host, port, peers, apiHost, apiPort, monitoringPort, chainConfig, ledgerConfig} <- identifyNode <$> parseHydraOptions
-  env@Environment{party} <- initEnvironment o
-  withTracer verbosity $ \tracer' ->
-    withMonitoring monitoringPort tracer' $ \tracer -> do
-      eq <- createEventQueue
-      withChain tracer party (putEvent eq . OnChainEvent) chainConfig $ \oc ->
-        withNetwork (contramap Network tracer) host port peers (putEvent eq . NetworkEvent) $ \hn ->
-          withAPIServer apiHost apiPort party (contramap APIServer tracer) (putEvent eq . ClientEvent) $ \server -> do
-            withCardanoLedger ledgerConfig $ \ledger -> do
-              node <- createHydraNode eq hn ledger oc server env
-              runHydraNode (contramap Node tracer) node
```

Comment:
This function was slightly refactored and moved below as `run`. The long line with the 'Options' assignment was split into smaller lines, closer to where the options get used.